### ISSUE TITLE
Parallelize completions from repl and LSP

### ISSUE THREAD

**Issue by Cyrik** [BetterThanTomorrow/calva#1680]
#1596 introduced the merging of nrepl and LSP completions. It's currently running one after the other, we should parallelize those calls.

**Comment by corasaurus-hex** [comment]
`Promise.any()` to the rescue!

**Comment by PEZ** [comment]
It's a bit more complicated than that. Firstly, the VS Code API does not allow us to first deliver one answer, then change our mind. But even if it would, Promise.any() gives us the winning answer. We want that first, then when the second answer is in, we want to reconcile the two based on the config. It's more like a generator with a reconcile step when the generator is done.

**Comment by Cyrik** [comment]
You are right. But speeding up the merge case is easier, because we can use promise.all.
The no-merge case can probably be sped up by creating all promises first, but still waiting on them in order.

**Comment by PEZ** [comment]
Indeed, for the merge it makes sense (with `Promise.all()`). Still even then, we would like to have the option to first provide the non-merged quickest answer and then change our mind and provide the merged one. But we should probably tease out some lower hanging fruit than that, as you are suggesting.

I'm thinking that we are starting to see a few more strategies now when we have two use cases in place, given the constraints of the VS Code API.

* For some cases we think it's worth waiting for the merge. We can use `Promise.all()` here. The priority array tells us which one should be merged onto the other.
* For some cases we prefer one over the other, but still think we can afford to wait for the slowest. Definitions comes to mind, where dynamic info can be out of whack with what's on file. `Promise.all()` and pick the one we prefer, if there is a choice.
* For some case we prefer one over the other and don't want to wait for both. Signatures comes to mind, where our nrepl solution is both noticeably quicker and better than the lsp provided one. Ask nrepl first, then lsp if nrepl doesn't provide something.
* For some cases we might not care, we return the first answer, `Promise.any()`.
